packages/hash/frontend/src/middleware/return-types-as-json/generate-query-args.ts

```diff
@@ -0,0 +1,69 @@
+import { VersionedUri } from "@blockprotocol/type-system";
+import { DocumentNode } from "graphql";
+import {
+  GetDataTypeQueryVariables,
+  GetEntityTypeQueryVariables,
+  GetPropertyTypeQueryVariables,
+} from "../../graphql/apiTypes.gen";
+import { getDataTypeQuery } from "../../graphql/queries/ontology/data-type.queries";
+import { getEntityTypeQuery } from "../../graphql/queries/ontology/entity-type.queries";
+import { getPropertyTypeQuery } from "../../graphql/queries/ontology/property-type.queries";
+
+/**
+ * Return the internal query string from a gql-tagged query, i.e. gql`string` -> string
+ */
+const queryStringFromNode = (node: DocumentNode) => {
+  const string = node.loc?.source.body.toString();
+
+  if (!string) {
+    throw new Error("Node did not contain 'loc'");
+  }
+
+  return string;
+};
+
+const zeroDepth = { outgoing: 0 };
+
+type OntologyType = "data" | "entity" | "property";
```

Comment:
I have a feeling we may have this sort of enum somewhere else, maybe in another form, but may be in the subgraph lib?

Reply:
@TimDiekmann Any idea?

Reply:
I can add it if we find it, although string unions are a bit easier to deal with than enums so would prefer the former if we're writing things ourselves

Reply:
We at least have `OntologyVertex` in the generated `api.ts`, which has a kind of unstable name for `OntologyVertexOneOf.kind`, so I guess we cannot use that.
We also have `OntologyVertex` defined in the subgraph library:
```ts
export type OntologyVertex =
  | DataTypeVertex
  | PropertyTypeVertex
  | EntityTypeVertex;

export type DataTypeVertex = { 
  kind: "dataType"; 
  inner: DataTypeWithMetadata 
};

export type PropertyTypeVertex = {
  kind: "propertyType";
  inner: PropertyTypeWithMetadata;
};

export type EntityTypeVertex = {
  kind: "entityType";
  inner: EntityTypeWithMetadata;
};
```

I'm not experienced enough with type definitions in TS, but maybe we can use the latter?

At least I'd prefer to be consistent with the strings, so change them to contain `type` as well (if possible)
```suggestion
type OntologyType = "dataType" | "entityType" | "propertyType";
```

Reply:
We cannot use `dataType` etc as these are extracted from the URL, e.g. `https://example.com/types/[something]-type`. `SchemaKind` in that linked file would be closest but it is not exported. 

I'm going to merge this for now but if there's some other suitable type please let me know and I'll update it. This will only be an issue if we ever add or change the name of an existing type though.

Reply:
I don't think we have a `kebab-case` enum anywhere, so this seems fine to me. I agree with Tim though, I really dislike us having `"data" | "entity" | "property"` in code when talking about types, it's reallyyy confusing.

> We cannot use `dataType` etc as these are extracted from the URL, e.g. `https://example.com/types/[something]-type`. 

Can we not extract it like `https://example.com/types/[something]/` where `something` is `entity-type` for example?

Reply:
Sure, I can do this as a driveby. I think the potential for confusion is pretty limited in this case as it's localized to one file and called `OntologyType` but happy to clean it up later.

Reply:
Drive-by at some point would be good I think. I appreciate the name of the type is clear, but the issue is that type can be used in bigger objects, and then you inevitably end up seeing other objects like

```
updateStateArgs = {
  type: "entity",
  ...
}
```

Because `type` was `OntologyType`